src/ng/compile.js

```diff
@@ -1625,6 +1630,27 @@ function $CompileProvider($provide, $$sanitizeUriProvider) {
       };
     }
 
+
+    function compileTemplate($templateNodes, parentTranscludeFn, terminalPriority, ignoreDirective, previousCompileContext) {
+      var defaultSlot = [];
+      var slots = createMap();
+      forEach($templateNodes, function(node) {
+        var slotName = jqLite(node).attr('ng-transclude-slot');
```

Comment:
Yes, this is a bit of a problem to do efficiently since at this stage we haven't extracted the attributes, so we would need to iterate over all the attributes or over all the possible variations of `ng-transclude-slot`.